pandas/core/categorical.py

```diff
@@ -1735,31 +1737,34 @@ def convert_categorical(x):
             return x.get_values()
         return x.ravel()
 
-    typs = get_dtype_kinds(to_concat)
-    if not len(typs-set(['object','category'])):
-
-        # we only can deal with object & category types
-        pass
-
-    else:
-
+    if get_dtype_kinds(to_concat) - set(['object', 'category']):
```

Comment:
so i think you're pointing out something that i'm not addressing:

```python
In [4]: s = pd.Series(list('aabbcd'))

In [5]: s2 = pd.Series(list('aabb')).astype('category')

In [6]: pd.concat([s,s2])
Out[6]:
0      a
1      a
2      b
3      b
4    NaN
5    NaN
0      a
1      a
2      b
3      b
dtype: category
Categories (2, object): [a, b]
```

*However*, that was run on current master and is a separate issue from the performance of concatenation.